evm_arithmetization/src/cpu/kernel/asm/core/util.asm

```diff
@@ -12,6 +12,17 @@
     // stack: (empty)
     %mload_global_metadata(@GLOBAL_METADATA_LARGEST_CONTEXT)
     %add_const(0x10000000000000000) // scale each context by 2^64
+    // stack: new_ctx
+    DUP1
+
+    // Memory addresses are represented as `ctx.2^64 + segment.2^32 + offset`,
+    // each address component expected to fit in a 32-bit limb.
+    // We enforce here that the new context id won't overflow.
+    PUSH 0xffffffffffffffffffffffff // 2^96 - 1
+    // stack: max, new_ctx, new_ctx
+    LT
+    %jumpi(fault_exception)
```

Comment:
Since this cannot happen in practice and would either be a bug or a malicious run, we might as well panic instead.

Reply:
That would go against their claims of following the exact YP specifications. Nothing prevents the block gas limit to be an arbitrary U256 value (although, I agree, we won't ever have it and it would break our other assumptions).

Reply:
Note that `%ensure_reasonable_offset` doesn't PANIC but throws a `fault_exception` too